R/compute_duration.R

```diff
@@ -157,7 +187,13 @@ compute_duration <- function(start_date,
   }
 
   # derive the duration in the output unit
-  duration <- time_length(start_date %--% end_date, unit = out_unit)
+  duration <-
+    switch(type,
```

Comment:
YES!!! `switch` I love this function!!  So much easier to understand.  @pharmaverse/admiral a good function to keep in your back pocket!

![image](https://github.com/pharmaverse/admiral/assets/10111024/688c78f8-4d14-428a-ba4e-3f8ddd869ba2)